### PR TITLE
chore(argo-workflows): update comment for azure artifact repository blobNameFormat

### LINKED ISSUE #2200: Artifact Repository for Azure Blob - values/examples not reflecting AzureArtifiactRepository blobNameFormat
### Describe the bug

[Values](https://github.com/argoproj/argo-helm/blob/main/charts/argo-workflows/values.yaml#L753 ) and examples in the document for using Azure Blob artifact repository does not reflect [AzureArtifiactRepository](AzureArtifiactRepository). Following the document/examples leads to following error in the argo workflow controller: `Failed to register watch for controller config map: error unmarshaling JSON: while decoding JSON: json: unknown field \"blob\ in the workflow controller`

```yaml
...
artifactRepository:
  azure:
    endpoint: https://myblob.blob.core.windows.net
    container: mycontainer
    blob: myblob # this should be blobNameFormat
    accountKeySecret:
      name: my-azure-storage-credentials
      key: my-account-access-key
```
This started happening after upgrading to 0.32.1 but was in several version behind before upgrading. I am assuming this might have started complaining after https://github.com/argoproj/argo-workflows/blob/master/docs/upgrading.md#feat-fail-on-invalid-config-8295

Passing in `blobNameFormat` instead of `blob` seems to work

### Related helm chart

argo-workflows

### Helm chart version

0.32.1

### To Reproduce

Install Argo Workflow with Chart 0.32.1 and use azure artifactRepository with `blob` key

```
artifactRepository:
  azure:
    endpoint: https://myblob.blob.core.windows.net
    container: mycontainer
    blob: myblob # this should be blobNameFormat
    accountKeySecret:
      name: my-azure-storage-credentials
      key: my-account-access-key
```

### Expected behavior

This is probably expected behavior but will requires update in document

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jsong336**
Resolves #2200

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation) - no change with `./scripts/helm-docs.sh`
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default. - no backward compatible/default needed
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-workflows/templates/crds/argoproj.io_workflowtaskresults.yaml [hunk 1]
```diff
@@ -116,7 +116,7 @@ spec:
                           required:
                           - key
                           type: object
-                        blob:
+                        blobNameFormat:
```
Comment (jmeridth) [inline charts/argo-workflows/templates/crds/argoproj.io_workflowtaskresults.yaml:119]: The upstream CRD still has `blob` [here](https://github.com/argoproj/argo-workflows/blob/cd2e4e564960d6c5f1a772d9d27672a08b3a6bcf/manifests/base/crds/minimal/argoproj.io_workflowtaskresults.yaml#L111-L112).  Please let me know where you are seeing the `blobNameFormat` in v3.4.10 upstream.

Comment (jmeridth) [reply charts/argo-workflows/templates/crds/argoproj.io_workflowtaskresults.yaml:119]: Are we waiting on an upstream change I'm missing?  I've looked through [here](https://github.com/search?q=repo%3Aargoproj%2Fargo-workflows+blobNameFormat&type=code) and didnt' find anything.

Comment (jsong336) [reply charts/argo-workflows/templates/crds/argoproj.io_workflowtaskresults.yaml:119]: sorry I am absolute novice here. I was looking at doc [here](https://github.com/argoproj/argo-workflows/blob/cd2e4e564960d6c5f1a772d9d27672a08b3a6bcf/sdks/java/client/docs/IoArgoprojWorkflowV1alpha1AzureArtifactRepository.md?plain=1#L12), and also had issue from #2200.

Comment (jsong336) [reply charts/argo-workflows/templates/crds/argoproj.io_workflowtaskresults.yaml:119]: oops, sorry I think I was confusing AzureArtifactRepository and AzureArtifact. No need for this line

File: charts/argo-workflows/Chart.yaml [hunk 2]
```diff
@@ -18,3 +18,5 @@ annotations:
   artifacthub.io/changes: |
     - kind: added
       description: add imagePullSecret for workflow sa
```
Comment (yu-croco) [inline charts/argo-workflows/Chart.yaml:20]: @jsong336 
Please remove old one. 🙋

File: charts/argo-workflows/values.yaml [hunk 3]
```diff
@@ -752,7 +752,7 @@ artifactRepository:
   azure: {}
   # endpoint: https://mystorageaccountname.blob.core.windows.net
   # container: my-container-name
-  # blob: path/in/container
+  # blobNameFormat: path/in/container
```
Comment (yu-croco) [inline charts/argo-workflows/values.yaml:755]: LGTM.
https://argoproj.github.io/argo-workflows/fields/
image

File: charts/argo-workflows/Chart.yaml [hunk 4]
```diff
@@ -18,3 +18,5 @@ annotations:
...
+    - kind: updated
```
Comment (yu-croco) [inline charts/argo-workflows/Chart.yaml:21]: Probably `updated` is not supported in `artifacthub.io/changes` , so maybe `fixed` ?

Ref: https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog

image

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution. LGTM.

**Comment by jsong336** [comment]
> Thank you for your contribution. LGTM.

no thanks for reviewing. hopefully I could contribute more in future

**Review by jmeridth — APPROVED** [review]

**Pull request merged**